package.json

```diff
@@ -12,6 +12,11 @@
     "@mux/mux-node": "^3.0.4",
     "@mux/upchunk": "^2.2.2",
     "@mux/videojs-kit": "^0.2.0",
+    "@strapi/design-system": "^0.0.1-alpha.71",
+    "@strapi/helper-plugin": "^4.0.4",
+    "@strapi/icons": "^0.0.1-alpha.71",
+    "@strapi/parts": "^0.0.1-alpha.50",
```

Comment:
Hi @erikpena , The issue is here. You have `@strapi/helper plugin`, but this will also be imported in the parent strapi app's `node_modules`. Since you have this dependency installed here it will be used instead of the one in the parent Strapi app. Somewhere along the line it doesn't grab the right context provider and that is why the useRBACProvider hook is returning null.

You can delete these packages from your package.json and run yarn. Then the `<CheckpagePermisions>` component should work.

Hope that helps. Keep me posted. I am going to update the docs, migration guide, and plugin generators.

Reply:
Aha, okay.  That makes complete sense.  It always feels weird not installing dependencies that I'm actually importing into my project, but I'll make these changes.  Thank you and please extend my appreciation to whomever you paired up with.

Reply:
Hey @markkaylor, yea we are having dependency issues here.  When I remove the dependencies that you mentioned above, then Strapi throws the following errors while it is creating the builds—

```strapi_1    | ModuleNotFoundError: Module not found: Error: Can't resolve '@strapi/helper-plugin' in '/srv/strapi-plugin-mux-video-uploader/admin/src'
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/Compilation.js:2011:28
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:795:13
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:275:22
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:431:22
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:124:11
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:667:25
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:852:8
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:972:5
strapi_1    |     at /srv/strapi/app/node_modules/neo-async/async.js:6883:13
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:955:45
strapi_1    |     at finishWithoutResolve (/srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:296:11)
strapi_1    |     at /srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:362:15
strapi_1    |     at /srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:410:5
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
strapi_1    | resolve '@strapi/helper-plugin' in '/srv/strapi-plugin-mux-video-uploader/admin/src'
strapi_1    |   Parsed request is a module
strapi_1    |   using description file: /srv/strapi-plugin-mux-video-uploader/package.json (relative path: ./admin/src)
strapi_1    |     Field 'browser' doesn't contain a valid alias configuration
strapi_1    |     resolve as module
strapi_1    |       /srv/strapi-plugin-mux-video-uploader/admin/src/node_modules doesn't exist or is not a directory
strapi_1    |       /srv/strapi-plugin-mux-video-uploader/admin/node_modules doesn't exist or is not a directory
strapi_1    |       looking for modules in /srv/strapi-plugin-mux-video-uploader/node_modules
strapi_1    |         single file module
strapi_1    |           using description file: /srv/strapi-plugin-mux-video-uploader/package.json (relative path: ./node_modules/@strapi/helper-plugin)
strapi_1    |             no extension
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |             .js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.js doesn't exist
strapi_1    |             .jsx
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.jsx doesn't exist
strapi_1    |             .react.js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.react.js doesn't exist
strapi_1    |         /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |       /srv/node_modules doesn't exist or is not a directory
strapi_1    |       /node_modules doesn't exist or is not a directory
strapi_1    |       looking for modules in /srv/strapi/app/node_modules/@strapi/admin/node_modules
strapi_1    |         single file module
strapi_1    |           using description file: /srv/strapi/app/node_modules/@strapi/admin/package.json (relative path: ./node_modules/@strapi/helper-plugin)
strapi_1    |             no extension
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |             .js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.js doesn't exist
strapi_1    |             .jsx
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.jsx doesn't exist
strapi_1    |             .react.js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.react.js doesn't exist
strapi_1    |         /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    | ModuleNotFoundError: Module not found: Error: Can't resolve '@strapi/helper-plugin' in '/srv/strapi-plugin-mux-video-uploader/admin/src'
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/Compilation.js:2011:28
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:795:13
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:275:22
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:431:22
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:124:11
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:667:25
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:852:8
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:972:5
strapi_1    |     at /srv/strapi/app/node_modules/neo-async/async.js:6883:13
strapi_1    |     at /srv/strapi/app/node_modules/webpack/lib/NormalModuleFactory.js:955:45
strapi_1    |     at finishWithoutResolve (/srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:296:11)
strapi_1    |     at /srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:362:15
strapi_1    |     at /srv/strapi/app/node_modules/enhanced-resolve/lib/Resolver.js:410:5
strapi_1    |     at eval (eval at create (/srv/strapi/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
strapi_1    | resolve '@strapi/helper-plugin' in '/srv/strapi-plugin-mux-video-uploader/admin/src'
strapi_1    |   Parsed request is a module
strapi_1    |   using description file: /srv/strapi-plugin-mux-video-uploader/package.json (relative path: ./admin/src)
strapi_1    |     Field 'browser' doesn't contain a valid alias configuration
strapi_1    |     resolve as module
strapi_1    |       /srv/strapi-plugin-mux-video-uploader/admin/src/node_modules doesn't exist or is not a directory
strapi_1    |       /srv/strapi-plugin-mux-video-uploader/admin/node_modules doesn't exist or is not a directory
strapi_1    |       looking for modules in /srv/strapi-plugin-mux-video-uploader/node_modules
strapi_1    |         single file module
strapi_1    |           using description file: /srv/strapi-plugin-mux-video-uploader/package.json (relative path: ./node_modules/@strapi/helper-plugin)
strapi_1    |             no extension
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |             .js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.js doesn't exist
strapi_1    |             .jsx
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.jsx doesn't exist
strapi_1    |             .react.js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin.react.js doesn't exist
strapi_1    |         /srv/strapi-plugin-mux-video-uploader/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |       /srv/node_modules doesn't exist or is not a directory
strapi_1    |       /node_modules doesn't exist or is not a directory
strapi_1    |       looking for modules in /srv/strapi/app/node_modules/@strapi/admin/node_modules
strapi_1    |         single file module
strapi_1    |           using description file: /srv/strapi/app/node_modules/@strapi/admin/package.json (relative path: ./node_modules/@strapi/helper-plugin)
strapi_1    |             no extension
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    |             .js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.js doesn't exist
strapi_1    |             .jsx
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.jsx doesn't exist
strapi_1    |             .react.js
strapi_1    |               Field 'browser' doesn't contain a valid alias configuration
strapi_1    |               /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin.react.js doesn't exist
strapi_1    |         /srv/strapi/app/node_modules/@strapi/admin/node_modules/@strapi/helper-plugin doesn't exist
strapi_1    | error Command failed with exit code 1.
strapi_1    | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.```

To be clear, this is being ran in Docker and I'm enabling this plugin as local deployment using the `plugins.js` config file—

```
// ./app/config/plugins.js
module.exports = ({ env }) => ({
  "mux-video-uploader": {
    enabled: true,
    resolve: '/srv/strapi-plugin-mux-video-uploader',
    config: {
      // user plugin config goes here
    },
  }
});
```